glean-core/ios/Glean/Scheduler/PingUploadOperation.swift

```diff
@@ -34,8 +38,23 @@ class PingUploadOperation: GleanOperation {
             return
         }
 
-        // Create the data task with appropriate status code handling
-        dataTask = URLSession.shared.dataTask(with: request) { _, response, error in
+        // Build a URLSession with no-caching suitable for uploading our pings
+        let config = URLSessionConfiguration.default
+        config.requestCachePolicy = NSURLRequest.CachePolicy.reloadIgnoringLocalCacheData
+        config.urlCache = nil
+        let session = URLSession(configuration: config)
+
+        // This asks the OS for more time when going to background in order to allow for background
+        // uploading of the pings.
+        backgroundTaskId = UIApplication.shared.beginBackgroundTask(withName: "Glean Upload Task") {
```

Comment:
Probably a stupid question: is it common/are we ok to have a background task name with spaces in it?

Reply:
It's really just a string "tag" and so the spaces are harmless. I'd also be okay renaming it to whatever you feel is more appropriate.

Reply:
No problem, it's fine!